### PR TITLE
SIM-DIGI-RECO-Validation era based customizations for GEM/ME0

### PR DESCRIPTION
**Pull request by dildick**
Era based customizations for GEM/ME0 SIM-DIGI-RECO (local/global) step. Customizations are derived from 
https://github.com/calabria/cmssw/blob/muonSystemAging/SLHCUpgradeSimulations/Configuration/python/gemCustoms.py and https://github.com/calabria/cmssw/blob/muonSystemAging/SLHCUpgradeSimulations/Configuration/python/me0Customs.py#L86

@calabria, @jshlee

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: IOMC/RandomEngine/python/IOMC_cff.py [hunk 1]
```diff
@@ -169,4 +169,13 @@
 
 randomEngineStateProducer = cms.EDProducer("RandomEngineStateProducer")
 
-
+from Configuration.StandardSequences.Eras import eras
+if eras.phase2_muon.isChosen() or eras.phase2dev_muon.isChosen():
```
Comment (Dr15Jones) [inline IOMC/RandomEngine/python/IOMC_cff.py]: I'm afraid this is not the appropriate way to use eras. We no longer want to support `isChosen()`. See https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras for documentation.

Comment (dildick) [inline IOMC/RandomEngine/python/IOMC_cff.py]: No problem, I'll use customization functions instead.

**Comment by slava77** [comment]
-1
based on comments from @Dr15Jones

**Comment by dildick** [comment]
* isChosen() has been replaced by customization functions
* Redundant packages in BuildFile.xml of SimMuon/GEMDigitizer were removed
* Code compiles

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by dildick** [comment]
* Bugfix in theMixObjects was fixed
* GEM/ME0 SimHits added to trackingParticles and trackingParticleSelection
* Code compiles

**Comment by dildick** [comment]
Last bits and pieces for now. To summarize: this PR is a comprehensive reform of the customizations of the SIM,DIGI,RECO and Validation step for the GEM/ME0 systems in the upgrade eras Phase2 and Phase2_dev. The list of packages affected by this PR is given below:
* SimGeneral/MixingModule
* SimGeneral/TrackingAnalysis
* IOMC/RandomEngine
* SimMuon/GEMDigitizer
* SimMuon/MCTruth
* SimMuon/Configuration
* RecoLocalMuon/Configuration
* RecoMuon/Configuration
* Validation/MuonGEMHits
* Validation/MuonGEMDigis
* Validation/MuonGEMRecHits
* Validation/MuonME0Validation
* Validation/Configuration

**Comment by Dr15Jones** [comment]
+1

**Comment by slava77** [comment]
@dildick 
- Phase2dev includes phase2dev_muon and also Phase2, which in turn includes phase2_muon. So, Phase2dev includes phase2_muon by construction. You have added identical modifications for both phase2_muon and phase2dev_muon. This is redundant. Only not-yet-working pieces should be added to phase2dev_muon, while the working parts should be in phase2_muon.
- it's probably time to cleanup gemCustoms.py, especially if corresponding eras are chosen so that the customs do not interfere
- GEM and ME0 customs are applied simultaneously. It makes more sense to put GEMs separately so that corresponding GEM-only modifiers can be applied to (post-LS2) Run3 era (whenever it gets defined) .. maybe even 2017 will get it as well if slice test goes well.

For a future PR, probably: .dat files for dead/masked channels to not seem appropriate for GEMs: plan to use database. There are many more channels in GEMs than in RPC. So, even a small fraction will require larger files than for RPC, which is not really maintainable.

**Comment by dildick** [comment]
@slava77 
   * Indeed, you're right. I'll remove the redundant lines
   * Ok, I'll clean-up the customizations in gemCustoms/me0Customs
   * Since Phase2 is targeting the Muon TDR with all upgrade muon systems in place, I did not consider splitting up GEM and ME0. I can split them up as soon a PostLS2 (2019) era is defined. 

About the .dat files. I'll check with GEM DPG what the plans are.

**Comment by dildick** [comment]
Customizations in SLHCUpgradeSimulations/Configuration are largely removed. Only GEM HLT remains for now.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by cvuosalo** [comment]
+1

For #13380 59acfb0419b0ea6b5c9b9438e72831a571626eb8

Adding SIM/DIGI/RECO/Validation era-based customizations for GEM/ME0. There should be no change in standard workflows.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-04-03-1100 show no significant differences, as expected.

**Comment by deguio** [comment]
+1

**Comment by davidlange6** [comment]
@dildick - can we modify the code to take the .dat files as an configuration option instead of needing an empty file to mean no dead channels (etc)- is strange to have empty files in this PR.

**Comment by dildick** [comment]
Code with masked/dead strips appears to be commented out in GEMRecHitProducer.cc. Empty .dat files can thus be removed.

**Comment by civanch** [comment]
please test

**Comment by davidlange6** [comment]
Thanks!

**Pull request merged**